include/pybind11/numpy.h

```diff
@@ -129,6 +129,11 @@ struct npy_api {
         NPY_STRING_, NPY_UNICODE_, NPY_VOID_
     };
 
+    typedef struct {
+        const Py_intptr_t *ptr;
```

Comment:
I think it's safe to make ptr `const` here instead of having `const_cast<Py_intptr_t*>(new_shape->data()` later in resize()